client/utils/serde/clubs-and-events.ts

```diff
@@ -0,0 +1,82 @@
+import moment from 'moment-timezone'
+import { Club, Event } from '../../utils'
+
+// TODO: remove these duplicated types from the backend with shared api types
+export interface BasicEventDTO {
+  name: string
+  startTime: string
+  endTime: string
+  bannerImage?: string
+  tags: Array<string>
+}
+
+export interface ClubDTO {
+  name: string
+  description: string
+  links: {
+    bannerImage: string
+    iconImage: string
+    facebook?: string
+    twitter?: string
+    instagram?: string
+    website?: string
+  }
+  tags: Array<string>
+  events: Array<BasicEventDTO>
+}
+
+interface DeserializedClubsAndEvents {
+  deserializedClubs: Array<Club>
+  deserializedEvents: Array<Event>
+}
+
+export const deserializeClubsAndEvents = (
+  serializedClubs: Array<ClubDTO> | undefined
+): DeserializedClubsAndEvents => {
+  if (!serializedClubs) {
+    return {
+      deserializedClubs: [],
+      deserializedEvents: [],
+    }
+  }
+
+  let eventId = 0
+  const deserializedEvents: Array<Event> = []
+
+  const deserializedClubs = serializedClubs.map((club, clubId) => {
+    return {
+      ...club,
+      id: clubId,
+      events: club.events.map((event) => {
+        const deserializedEvent = deserializeEvent(event, eventId++, club)
```

Comment:
so `eventId` seems to always be equal to `deserializedEvents.length`, maybe we can just use that?
```suggestion
        const deserializedEvent = deserializeEvent(event, deserializedEvents.length, club)
```

Reply:
you're a smart cs mans justin